test_common/harness/testHarness.c

```diff
@@ -891,3 +900,29 @@ cl_device_id GetOpposingDevice( cl_device_id device )
     // Should never get here
     return NULL;
 }
+
+Version get_device_cl_version(cl_device_id device)
+{
+    size_t str_size;
+    cl_int err = clGetDeviceInfo(device, CL_DEVICE_VERSION, 0, NULL, &str_size);
+    assert(err != CL_SUCCESS && "Could not get CL_DEVICE_VERSION size.");
```

Comment:
wrong expression, should be err == CL_SUCCESS

Reply:
Yep, silly mistake. This actually shows that we need something better than assert, as it doesn't catch it in release build.